pkg/cloud/gcp/bigqueryquerier.go

```diff
@@ -49,9 +50,12 @@ func (bqq *BigQueryQuerier) Query(ctx context.Context, queryStr string) (*bigque
 	// If result is empty and connection status is not already successful update status to missing data
 	if iter == nil && bqq.ConnectionStatus != cloud.SuccessfulConnection {
 		bqq.ConnectionStatus = cloud.MissingData
-		return iter, nil
+	} else {
+		bqq.ConnectionStatus = cloud.SuccessfulConnection
```

Comment:
For my understanding, how does this prevent the function from returning a nil iterator?

The downstream problem was returning a nil iterator with no error, which still seems to be a possibility here.

Reply:
For context this line previously read
```
return query.Read(ctx)
}
```
I did it like this because it closely mirrors what it was before